Documentation/Design/Lambdas/Lambda Declare.md

```diff
@@ -0,0 +1,162 @@
+# Lambda Declare
+
+## Overview
+GLS will support declaring lambda functions. In order to declare a function that takes a lambda function as a parameter, the lambda must first be declared.
+
+## Commands
+
+### `lambda declare`
+`lambda declare : lambdaTypeName lambdaFunctionName returnType *`[parameterName parameterType ...]`*
```

Comment:
I'm kind of split on the commas. If we put them in, it almost looks like the GLS command is supposed to have commas, which isn't the case. In the real command, they're just space-separated.